lib/resourcemerge/core_test.go

```diff
@@ -113,6 +114,138 @@ func TestEnsurePodSpec(t *testing.T) {
 				Containers: []corev1.Container{
 					corev1.Container{Name: "test"}}},
 		},
+		{
+			name: "remove limits and requests on container",
+			existing: corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{
+						Name: "test",
+						Resources: corev1.ResourceRequirements{
+							Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2m")},
+							Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2m")},
+						},
+					},
+				},
+			},
+			input: corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{Name: "test"}}},
+
+			expectedModified: true,
+			expected: corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{Name: "test"},
+				},
+			},
+		},
+		{
+			name: "modify limits and requests on container",
+			existing: corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{
+						Name: "test",
+						Resources: corev1.ResourceRequirements{
+							Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2m")},
+							Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("2m")},
+						},
+					},
+				},
+			},
+			input: corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{
+						Name: "test",
+						Resources: corev1.ResourceRequirements{
+							Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("4m")},
+							Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("4m")},
+						},
+					},
+				},
+			},
+
+			expectedModified: true,
+			expected: corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{
+						Name: "test",
+						Resources: corev1.ResourceRequirements{
+							Limits:   corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("4m")},
+							Requests: corev1.ResourceList{corev1.ResourceCPU: resource.MustParse("4m")},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "match limits and requests on container",
```

Comment:
@abhinavdahiya added matching existing and input last night with `expectedModified: false`